compass/validate.py

```diff
@@ -58,38 +58,68 @@ def compare_variables(test_case, variables, filename1, filename2=None,
         Whether to check to make sure files are valid outputs of steps in
         the test case.  This should be set to ``False`` if comparing with an
         output of a step in another test case.
+
+    skip_if_step_not_run : bool, optional
+        Whether to skip the variable comparison if a user did not run one (or
+        both) of the steps involved in the comparison.  This would happen if
+        users are running steps individually or has edited ``steps_to_run``
+        in the config file to exclude one of the steps.
     """
     work_dir = test_case.work_dir
 
-    if test_case.validation is not None:
-        validation = test_case.validation
-    else:
-        validation = {'internal_pass': None,
-                      'baseline_pass': None}
-
     path1 = os.path.abspath(os.path.join(work_dir, filename1))
     if filename2 is not None:
         path2 = os.path.abspath(os.path.join(work_dir, filename2))
     else:
         path2 = None
-    if check_outputs:
-        file1_found = False
-        file2_found = False
-        for step_name, step in test_case.steps.items():
-            for output in step.outputs:
-                # outputs are already absolute paths combined with the step dir
-                if output == path1:
-                    file1_found = True
-                if output == path2:
-                    file2_found = True
 
+    file1_found = False
+    file2_found = False
+    step_name1 = None
+    step_name2 = None
+    for step_name, step in test_case.steps.items():
+        for output in step.outputs:
+            # outputs are already absolute paths combined with the step dir
+            if output == path1:
+                file1_found = True
+                step_name1 = step_name
+            if output == path2:
+                file2_found = True
+                step_name2 = step_name
+
+    if check_outputs:
         if not file1_found:
             raise ValueError('{} does not appear to be an output of any step '
                              'in this test case.'.format(filename1))
         if filename2 is not None and not file2_found:
             raise ValueError('{} does not appear to be an output of any step '
                              'in this test case.'.format(filename2))
 
+    if skip_if_step_not_run:
+        step1_not_run = (file1_found and
+                         step_name1 not in test_case.steps_to_run)
```

Comment:
If a step is run manually (`compass run` from the step dir), does `steps_to_run` get updated  accordingly?  If not, this will not be accurate.

Reply:
Yes, when you run a single step, `steps_to_run` automatically gets set to a list with only that step in it, so this will work as expected.  This is also how things work now (before this PR), which is why all these test cases were checking to see if steps were in `steps_to_run`, just like is happening here.

Reply:
Ok, thanks - is that stored in the pickle file and then populated in the testcase object when `compass run` is run?

Reply:
I believe the pickle file stores the full list of `steps_to_run`, and it's here in the `run` module that it `steps_to_run` gets modified:
https://github.com/altheaden/compass/blob/master/compass/run.py#L224